src/Authentication/Authenticators/HMAC_SHA256.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Authentication\Authenticators;
+
+use CodeIgniter\HTTP\IncomingRequest;
+use CodeIgniter\I18n\Time;
+use CodeIgniter\Shield\Authentication\AuthenticationException;
+use CodeIgniter\Shield\Authentication\AuthenticatorInterface;
+use CodeIgniter\Shield\Entities\User;
+use CodeIgniter\Shield\Exceptions\InvalidArgumentException;
+use CodeIgniter\Shield\Models\TokenLoginModel;
+use CodeIgniter\Shield\Models\UserIdentityModel;
+use CodeIgniter\Shield\Models\UserModel;
+use CodeIgniter\Shield\Result;
+
+class HMAC_SHA256 implements AuthenticatorInterface
```

Comment:
This classname does not follow PSR-1.
HMACSha256? HmacSha256?

Reply:
I debated how to name this as the entire thing is a big acronym.  I will rework and update shortly.